src/Runner.Worker/ActionManifestManager.cs

```diff
@@ -55,22 +56,35 @@ public override void Initialize(IHostContext hostContext)
 
         public ActionDefinitionData Load(IExecutionContext executionContext, string manifestFile)
         {
-            var context = CreateContext(executionContext);
+            var templateContext = CreateContext(executionContext);
             ActionDefinitionData actionDefinition = new ActionDefinitionData();
+
+            // Clean up file name real quick
+            // Instead of using Regex which can be computationally expensive, 
+            // we can just remove the # of characters from the fileName according to the length of the basePath
+            string basePath = _hostContext.GetDirectory(WellKnownDirectory.Actions);
+            string fileName = manifestFile;
+            if (manifestFile.Length > basePath.Length + 1)
+            {
+                fileName = manifestFile.Remove(0, basePath.Length + 1);
+            }
+
             try
             {
                 var token = default(TemplateToken);
 
                 // Get the file ID
-                var fileId = context.GetFileId(manifestFile);
-                _fileTable = context.GetFileTable();
+                var fileId = templateContext.GetFileId(fileName);
+
+                // Add this file to the FileTable in executionContext
+                executionContext.FileTable.Add(fileName);
```

Comment:
After you add the "fileName" the `fileId` should match `executionContext.FileTable.Length`

Can we add an assert.

Reply:
only add if new